src/main/java/org/openrewrite/java/testing/junit5/ExpectedExceptionToAssertThrows.java

```diff
@@ -93,59 +116,128 @@ public String getDescription() {
 
             J.MethodDeclaration m = super.visitMethodDeclaration(methodDecl, ctx);
 
-            J.MethodInvocation expectedExceptionMethodInvocation = getCursor().pollMessage(EXPECT_INVOCATION_KEY);
-            if (expectedExceptionMethodInvocation != null) {
-                List<Expression> args = expectedExceptionMethodInvocation.getArguments();
+            J.MethodInvocation expectMethodInvocation = getCursor().pollMessage(EXPECT_INVOCATION_KEY);
+            J.MethodInvocation expectMessageMethodInvocation = getCursor().pollMessage(EXPECT_MESSAGE_INVOCATION_KEY);
+            J.MethodInvocation expectCauseMethodInvocation = getCursor().pollMessage(EXPECT_CAUSE_INVOCATION_KEY);
+            if (expectMethodInvocation == null &&
+                    expectMessageMethodInvocation == null &&
+                    expectCauseMethodInvocation == null) {
+                return m;
+            }
+
+            boolean isExpectArgAMatcher = false;
+            if (expectMethodInvocation != null) {
+                List<Expression> args = expectMethodInvocation.getArguments();
                 if (args.size() != 1) {
                     return m;
                 }
 
-                Expression expectedException = args.get(0);
-                JavaType.FullyQualified argType = TypeUtils.asFullyQualified(expectedException.getType());
-                if (argType == null || !argType.getFullyQualifiedName().equals("java.lang.Class")) {
+                final Expression expectMethodArg = args.get(0);
+                if (expectMethodArg instanceof J.MethodInvocation) {
+                    isExpectArgAMatcher = isHamcrestMatcher((J.MethodInvocation) expectMethodArg);
+                    if (!isExpectArgAMatcher) {
+                        return m;
+                    }
+                } else {
+                    final JavaType.FullyQualified argType = TypeUtils.asFullyQualified(expectMethodArg.getType());
+                    if (argType == null || !argType.getFullyQualifiedName().equals("java.lang.Class")) {
+                        return m;
+                    }
+                }
+            }
```

Comment:
In the above code, we return out of the method when the method invocation does not match what we are expecting. This will short-circuit any potential checks for the other two method invocations.

Reply:
Ya, it's designed to ensure the method arguments match the recipe expectations prior to making changes.
The ExpectedException methods args have remained the same, since it was added in '12.
So, technically, the recipe will refactor the current JUnit4 version and previous versions, but it's possible for changes to happen eventually.

I.E. there is an open issue to remove the dependency on Hamcrest matcher - they tried to deprecate it, but the deprecation was reverted.

I'm not sure how to handle those types of edge cases and couldn't find a precedence for future proofing recipes.
If you think it's worth while, I could add code to leave behind an artifact like a comment about why the recipe didn't refactor the test.

Reply:
I am good with how it works now, its much better than it was.